tests/common/grpc_test.go

```diff
@@ -0,0 +1,139 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	intf "go.etcd.io/etcd/tests/v3/framework/interfaces"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+func TestAuthority(t *testing.T) {
```

Comment:
What test are you migrating? Please note that TestAuthority is a little unique than other tests. 

There are 2 versions of it:
* https://github.com/etcd-io/etcd/blob/100a7b4b0dc6e5f098a837f68885fc14648074d2/tests/e2e/ctl_v3_grpc_test.go#L36
* https://github.com/etcd-io/etcd/blob/100a7b4b0dc6e5f098a837f68885fc14648074d2/tests/integration/grpc_test.go#L31

Reply:
Now, I'm trying to migrate both e2e and integration tests to `tests/common/grpc_test.go`.

Reply:
I’m not yet sure a straight migration will work, so my plan is to implement an initial port first and then evaluate feasibility based on what I learn.

Reply:
Think tests/e2e/ctl_v3_kv_test.go should be pretty easy.

Reply:
Yeah, other tests might be easier to migrate than this one.

I’ll push my latest changes and update the PR description later so you can give me some directional feedback.
If the approach feels too heavy or misaligned, I’ll mark this PR as pending and move on to easier migrations first.